state/client.go

```diff
@@ -0,0 +1,48 @@
+package state
+
+import (
+	"context"
+	"fmt"
+	"github.com/cloudquery/plugin-sdk/v4/plugin"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+const defaultMaxMsgSizeInBytes = 100 * 1024 * 1024 // 100 MiB
+
+type ConnectionOptions struct {
+	MaxMsgSizeInBytes int
+}
+
+// NewGrpcConnectedClient returns a state client and a gRPC connection to the state backend with a 100MiB max message size.
+// The state client is guaranteed to be non-nil (it defaults to the NoOpClient).
+func NewGrpcConnectedClient(ctx context.Context, backendOpts *plugin.BackendOptions) (Client, *grpc.ClientConn, error) {
+	return NewGrpcConnectedClientWithOptions(ctx, backendOpts, ConnectionOptions{MaxMsgSizeInBytes: defaultMaxMsgSizeInBytes})
+}
+
+// NewGrpcConnectedClientWithOptions returns a state client and a gRPC connection to the state backend.
+// The state client is guaranteed to be non-nil (it defaults to the NoOpClient).
+func NewGrpcConnectedClientWithOptions(ctx context.Context, backendOpts *plugin.BackendOptions, opts ConnectionOptions) (Client, *grpc.ClientConn, error) {
```

Comment:
IMO we should provide a nop closer instead of `*grpc.ClientConn` or something similar (& also make sure that the clients calling this method will run `close` manually.
Otherwise, this returned value has no sense

Reply:
Yeah returning a Closer is probably a good call